### PR TITLE
Content moderation with lancedb

### PR DESCRIPTION
**Pull request by mrubash1**

### CODE REVIEW

File: build-lancedb.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+from lancedb.embeddings import with_embeddings
+from sentence_transformers import SentenceTransformer
+
+# Inference and training function
+def embed_func(batch):
+    return [model.encode(sentence) for sentence in batch]
+
+
+# Training Functions
+def load_data(data_loading_path="training_data/all_in_one_jigsaw.csv"):
```
Comment (oaali92) [inline build-lancedb.py:16]: can we host this somewhere on AWS or GDrive and pull? Will make it much easier.

File: build-lancedb.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        dataset_path (str): The path to the dataset on Kaggle in the format 'user/dataset-name'.
+        download_path (str): The local path where the dataset will be downloaded and unzipped.
+    """
+    if not os.path.exists(os.path.join(os.getcwd(), data_loading_path)):
+        print("Downloading dataset from Kaggle %s to %s" % (dataset_path, download_path))
+        if not os.path.exists(download_path):
+            os.makedirs(download_path)
+
+        # Download the dataset
+        command = f"kaggle datasets download -d {dataset_path} -p {download_path}"
```
Comment (oaali92) [inline build-lancedb.py:82]: nvm!

**Review by oaali92 — APPROVED** [review]
Looks great!! Didn't try to run yet but will do so later this week.

File: build-lancedb.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    data = create_embeddings(df, row_limit=args.samples_of_jigsaw_to_process, show_progress=True)
+
+    tbl = create_lancedb_table(data, 
+        uri=args.lancedb_uri, 
+        name=args.lancedb_table_name, 
+        index_table=True, 
+        num_partitions=4)
+    
+    print ('Done!')
```
Comment (oaali92) [inline build-lancedb.py:141]: This build script is all you run - right?

**Review by oaali92 — APPROVED** [review]

File: build-lancedb.py (hunk 1 above)
Comment (mrubash1) [reply build-lancedb.py:16]: Great idea. Issue is that downloading Kaggle data requires registration - so we can't host the data elsewhere (at least in this public example)

**Pull request merged**